tests/integration/workflows/test_manual_merge_workflow.py

```diff
@@ -0,0 +1,112 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Tests for manual merge workflows."""
+
+from __future__ import absolute_import, division, print_function
+
+import json
+import os
+import pytest
+
+from invenio_workflows import workflow_object_class
+
+from inspirehep.modules.pidstore.minters import inspire_recid_minter
+from inspirehep.modules.records.api import InspireRecord
+from inspirehep.modules.workflows.tasks.merging import _read_wf_record_source
+from inspirehep.modules.workflows.workflows.manual_merge import start_merger
+from inspirehep.utils.record import get_source
+from inspirehep.utils.record_getter import get_db_record, RecordGetterError
+
+
+def read_file(test_dir, file_name):
+    base_dir = os.path.dirname(os.path.realpath(__file__))
+    with open(os.path.join(base_dir, test_dir, file_name)) as f:
+        return json.loads(f.read())
+
+
+def create_test_record(file_name):
+    rec = read_file('fixtures', file_name)
+    record = InspireRecord.create(rec)
+    inspire_recid_minter(str(record.id), record)
+    return record
+
+
+def _remove_uuid_authors(record):
+    for author in record['authors']:
+        if author.get('uuid'):
+            del author['uuid']
+    return record
+
+
+def test_manual_merge_existing_records(workflow_app):
+    head = create_test_record('manual_merge_record.json')
+    update = create_test_record('merger_update.json')
+
+    head_id = str(head.id)
+    update_id = str(update.id)
+
+    obj_id, merged, conflicts = start_merger(
+        head_id=head['control_number'],
+        update_id=update['control_number'],
+        current_user_id=1,
+        pid_type='lit'
+    )
+
+    # continue the workflow
+    obj = workflow_object_class.get(obj_id)
+
+    obj.continue_workflow('continue_next')
+
+    # retrieve it again because it has finished is job
+    obj = workflow_object_class.get(obj_id)
+    merged = obj.data
+
+    # since head and update have different acquisition sources
+    # they will be saved both as root
+    head_root = _read_wf_record_source(str(head_id), get_source(head))
+
+    assert head_root == head
+
+    update_root = _read_wf_record_source(str(update_id), get_source(update))
+    assert get_source(head) == get_source(update)
+    # since records have same source, only head is saved as root
+    assert update_root == {}
+
+    # check that head's content has been replaced by merged
+    new_head = get_db_record('lit', head['control_number'])
+    # new_head has a deleted_record field which is absent in merged var
+    del new_head['deleted_records']
```

Comment:
You probably want to first check that you have the right thing in `deleted_records` before deleting it and checking that the head has the same content of what you merged (this test can make a stronger assertion).